Mathlib/AlgebraicGeometry/Morphisms/Basic.lean

```diff
@@ -156,347 +206,256 @@ instance targetAffineLocally_respectsIso (P : AffineTargetMorphismProperty)
+
+theorem HasAffineProperty.of_iSup_eq_top
+    {ι} (U : ι → Y.affineOpens) (hU : ⨆ i, (U i : Opens Y) = ⊤)
+    (hU' : ∀ i, P.affineProperty (f ∣_ U i)) :
+    P f := by
+  rw [eq_targetAffineLocally P]
   classical
-  let S i : Y.affineOpens := ⟨_, isAffineOpen_opensRange (𝒰.map i)⟩
-  intro U
-  induction U using of_affine_open_cover S 𝒰.iSup_opensRange with
+  intro V
+  induction V using of_affine_open_cover U hU  with
   | basicOpen U r h =>
     haveI : IsAffine _ := U.2
-    have := hP.2 (f ∣_ U.1)
-    replace this := this (Y.presheaf.map (eqToHom U.1.openEmbedding_obj_top).op r) h
-    rw [← P.toProperty_apply] at this ⊢
-    exact (P.toProperty.arrow_mk_iso_iff (morphismRestrictRestrictBasicOpen f _ r)).mp this
+    have := AffineTargetMorphismProperty.IsLocal.to_basicOpen (f ∣_ U.1) ((Y.resTop U).inv r) h
+    exact (P.affineProperty.arrow_mk_iso_iff
+      (morphismRestrictRestrictBasicOpen f _ r)).mp this
   | openCover U s hs H =>
-    haveI : IsAffine _ := U.2
-    apply hP.3 (f ∣_ U.1) (s.image (Y.presheaf.map (eqToHom U.1.openEmbedding_obj_top).op))
-    · apply_fun Ideal.comap (Y.presheaf.map (eqToHom U.1.openEmbedding_obj_top.symm).op) at hs
-      rw [Ideal.comap_top] at hs
-      rw [← hs]
-      simp only [eqToHom_op, eqToHom_map, Finset.coe_image]
-      have : ∀ {R S : CommRingCat} (e : S = R) (s : Set S),
-          Ideal.span (eqToHom e '' s) = Ideal.comap (eqToHom e.symm) (Ideal.span s) := by
-        intro _ S e _
-        subst e
-        simp only [eqToHom_refl, CommRingCat.id_apply, Set.image_id']
-        -- Porting note: Lean didn't see `𝟙 _` is just ring hom id
-        exact (Ideal.comap_id _).symm
-      apply this
-    · rintro ⟨r, hr⟩
-      obtain ⟨r, hr', rfl⟩ := Finset.mem_image.mp hr
-      specialize H ⟨r, hr'⟩
-      rw [← P.toProperty_apply] at H ⊢
-      exact (P.toProperty.arrow_mk_iso_iff (morphismRestrictRestrictBasicOpen f _ r)).mpr H
-  | hU i =>
-    specialize h𝒰 i
-    rw [← P.toProperty_apply] at h𝒰 ⊢
-    exact (P.toProperty.arrow_mk_iso_iff (morphismRestrictOpensRange f _)).mpr h𝒰
-#align algebraic_geometry.target_affine_locally_of_open_cover AlgebraicGeometry.targetAffineLocally_of_openCover
-
-open List in
-theorem AffineTargetMorphismProperty.IsLocal.affine_openCover_TFAE
-    {P : AffineTargetMorphismProperty} (hP : P.IsLocal) {X Y : Scheme.{u}} (f : X ⟶ Y) :
-    TFAE
-      [targetAffineLocally P f,
-        ∃ (𝒰 : Scheme.OpenCover.{u} Y) (_ : ∀ i, IsAffine (𝒰.obj i)),
-          ∀ i : 𝒰.J, P (pullback.snd : (𝒰.pullbackCover f).obj i ⟶ 𝒰.obj i),
-        ∀ (𝒰 : Scheme.OpenCover.{u} Y) [∀ i, IsAffine (𝒰.obj i)] (i : 𝒰.J),
-          P (pullback.snd : (𝒰.pullbackCover f).obj i ⟶ 𝒰.obj i),
-        ∀ {U : Scheme} (g : U ⟶ Y) [IsAffine U] [IsOpenImmersion g],
-          P (pullback.snd : pullback f g ⟶ U),
-        ∃ (ι : Type u) (U : ι → Opens Y.carrier) (_ : iSup U = ⊤) (hU' : ∀ i, IsAffineOpen (U i)),
-          ∀ i, @P _ _ (f ∣_ U i) (hU' i)] := by
```

Comment:
I agree that it is more useful to have the equivalences as separate lemmas. Maybe this TFAE is still good as an overview of all characterizations though, but this might also just be a docstring.